tumblr/tumblr.js
# Start of typescript type definitions

This is a start of typescript type definitions. Closes #56 

## Testing
- checkout this branch in tumblr.js
- setup a new project in a new folder
```
yarn add typescript
yarn add ../tumblr.js # -- or whatever the path to your checked-out copy of tumblr.js is
tsc --init
```

- open [VS Code](https://code.visualstudio.com/)
- Create a new file index.ts
- [ ] see some intellisense as you're entering code
![image](https://user-images.githubusercontent.com/6033/35311266-a5546808-0083-11e8-8fb7-3cdc5103ba18.png)

```typescript
import * as tumblr from 'tumblr.js';

const client = tumblr.createClient({
    consumer_key: '...',
})

client.blogPosts('staff', (err, data) => {
    console.log(JSON.stringify(data, null, 2));
})
```
- compile to javascript: `tsc`
- run the script: `node index.js`
- [ ] see blog posts
```
$ node index.js
{ blog:
   { ask: false,
     ask_anon: false,
     ask_page_title: '',
     can_message: false,
...
```
